### ISSUE TITLE
Suppressing embeds in another user's message doesn't work anymore

### ISSUE THREAD

**Issue by muddyfish** [Rapptz/discord.py#6133]
## Summary

If you do `message.edit(suppress=True)` on a message not sent by you, you get the 'Cannot edit a message authored by another user' message

The issue is in [this piece of code](https://github.com/Rapptz/discord.py/blob/master/discord/message.py#L1006-L1023); it always adds a `allowed_mentions` field to the http request, which causes the issue, as the only thing allowed to be edited in another user's messages is the flags

## Reproduction Steps

```
await message.edit(suppress=True)
```
or 
```
await sent_message.edit(flags=MessageFlags.suppress_embeds.flag)
```

## Expected Results

Embeds in the message disappear

## Actual Results

> 403 Forbidden (error code: 50005): Cannot edit a message authored by another user

## Intents

Irrelevant 

## Checklist

- [x] I have searched the open issues for duplicates.
- [x] I have shown the entire traceback, if possible.
- [x] I have removed my token from display, if visible.
- [ ] I have provided the intents that my bot is using.

## System Information

Current HEAD (0995d077271ed97129c9d06833f710d2c2744dae)

**Comment by Zomatree** [comment]
There is a open pr for a fix for this #6132.

**Comment by Rapptz** [comment]
This is fixed now.
